crates/next-core/src/next_font_google/util.rs

```diff
@@ -0,0 +1,368 @@
+use std::cmp::Ordering;
+
+use anyhow::{anyhow, Context, Result};
+use indexmap::{indexset, IndexSet};
+
+use super::options::{FontData, FontWeights};
+
+#[derive(Debug, PartialEq)]
+pub struct FontAxes {
+    pub wght: IndexSet<String>,
+    pub ital: IndexSet<FontItal>,
+    pub variable_axes: Option<Vec<(String, String)>>,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum FontItal {
+    Italic,
+    Normal,
+}
+
+pub fn get_font_axes(
+    font_data: &FontData,
+    font_family: &str,
+    weights: &FontWeights,
+    styles: &IndexSet<String>,
+    selected_variable_axes: &Option<Vec<String>>,
+) -> Result<FontAxes> {
+    let all_axes = &font_data
+        .get(font_family)
+        .context("Font family not found")?
+        .axes;
+
+    let Some(defineable_axes) = all_axes else {
+        return Err(anyhow!("Font {} has no definable `axes`", font_family));
+    };
+
+    let has_italic = styles.contains("italic");
+    let has_normal = styles.contains("normal");
+    let ital = {
+        let mut set = IndexSet::new();
+        if has_normal {
+            set.insert(FontItal::Normal);
+        }
+        if has_italic {
+            set.insert(FontItal::Italic);
+        }
+        set
+    };
+
+    match weights {
+        FontWeights::Variable => {
+            if let Some(selected_variable_axes) = selected_variable_axes {
+                let definable_axes_tags = defineable_axes
+                    .iter()
+                    .map(|axis| axis.tag.to_owned())
+                    .collect::<Vec<String>>();
+
+                for tag in selected_variable_axes {
+                    if !definable_axes_tags.contains(tag) {
+                        return Err(anyhow!(
+                            "Invalid axes value {} for font {}.\nAvailable axes: {}",
+                            tag,
+                            font_family,
+                            definable_axes_tags.join(", ")
+                        ));
+                    }
+                }
+            }
+
+            let mut weight_axis = None;
+            let mut variable_axes = vec![];
+            for axis in defineable_axes {
+                if axis.tag == "wght" {
+                    weight_axis = Some(format!("{}..{}", axis.min, axis.max));
+                } else if let Some(selected_variable_axes) = selected_variable_axes {
+                    if selected_variable_axes.contains(&axis.tag) {
+                        variable_axes
+                            .push((axis.tag.clone(), format!("{}..{}", axis.min, axis.max)));
+                    }
+                }
+            }
+
+            let Some(weight_axis) = weight_axis else {
+                return Err(anyhow!("Expected wght axis to appear in font data for {}", font_family));
```

Comment:
A few variable fonts (like 3) don't have a wght axis. For example `Nabla`, maybe something like this:

```rust
    #[test]
    fn test_variable_font_without_wgth_axis() -> Result<()> {
        assert_eq!(
            get_stylesheet_url(
                "Roboto Serif",
                &FontAxes {
                    wght: indexset! {},
                    ital: indexset! {},
                    variable_axes: Some(vec![
                        ("EDPT".to_owned(), "0..200".to_owned()),
                        ("EHLT".to_owned(), "0..24".to_owned()),
                    ])
                },
                "optional"
            )?,
            "https://fonts.googleapis.com/css2?family=Nabla&display=optional"
        );

        Ok(())
    }
```